packages/core/primitives/event-dispatch/src/register_events.ts

```diff
@@ -6,34 +6,63 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import {EarlyEventContract, EarlyJsactionDataContainer} from './earlyeventcontract';
 import {EventContractContainer} from './event_contract_container';
 import {EventContract} from './eventcontract';
 
+type Container = {[key: string]: {[appId: string]: EarlyJsactionDataContainer}};
```

Comment:
nit: we can add a generic type here:

```suggestion
type Tracker<T> = {[key: string]: {[appId: string]: T}};
```
and use it in both functions in this file:

```
earlyJsactionTracker: Tracker<EventContract> = window,
```

(btw, should we rename this argument to something like `jsactionEventContractTracker` to drop "early" from it, since it's a different type)

and

```
earlyJsactionTracker: Tracker<EarlyJsactionDataContainer> = window,
```

Reply:
done